src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/AotTests.cs

```diff
@@ -112,6 +112,33 @@ public void BuildBasicApplicationReleaseProfiledAotWithoutDefaultProfile ()
 			StringAssertEx.DoesNotContainRegex (@$"Using profile data file.*{filename}\.aotprofile", b.LastBuildOutput, "Should not use default AOT profile", RegexOptions.IgnoreCase);
 		}
 
+		[Test]
+		[TestCase ("テスト", false, false, true)]
+		[TestCase ("テスト", true, true, false)]
+		[TestCase ("テスト", true, false, true)]
+		[TestCase ("随机生成器", false, false, true)]
+		[TestCase ("随机生成器", true, true, false)]
+		[TestCase ("随机生成器", true, false, true)]
+		[TestCase ("中国", false, false, true)]
+		[TestCase ("中国", true, true, false)]
+		[TestCase ("中国", true, false, true)]
+		public void BuildAotApplicationWithSpecialCharactersInProject (string testName, bool isRelease, bool aot, bool expectedResult)
+		{
+			if (!IsWindows)
+				expectedResult = true;
+			var rootPath = Path.Combine (Root, "temp", TestName);
+			var proj = new XamarinAndroidApplicationProject () {
+				ProjectName = testName,
+				IsRelease = isRelease,
+				AotAssemblies = aot,
+			};
+			proj.SetAndroidSupportedAbis ("armeabi-v7a",  "arm64-v8a", "x86", "x86_64");
+			using (var builder = CreateApkBuilder (Path.Combine (rootPath, proj.ProjectName))){
+				builder.ThrowOnBuildFailure = false;
+				Assert.AreEqual (expectedResult, builder.Build (proj), "Build should have succeeded.");
```

Comment:
Should this also verify the build error?

Reply:
done